src/sentry/monitors/endpoints/organization_monitor_index.py

```diff
@@ -266,56 +262,18 @@ def post(self, request: AuthenticatedHttpRequest, organization) -> Response:
         Create a new monitor.
         """
         validator = MonitorValidator(
-            data=request.data, context={"organization": organization, "access": request.access}
+            data=request.data,
+            context={"organization": organization, "access": request.access, "request": request},
         )
         if not validator.is_valid():
             return self.respond(validator.errors, status=400)
 
-        result = validator.validated_data
-
-        owner = result.get("owner")
-        owner_user_id = None
-        owner_team_id = None
-        if owner and owner.is_user:
-            owner_user_id = owner.id
-        elif owner and owner.is_team:
-            owner_team_id = owner.id
-
         try:
-            monitor = Monitor.objects.create(
-                project_id=result["project"].id,
-                organization_id=organization.id,
-                owner_user_id=owner_user_id,
-                owner_team_id=owner_team_id,
-                name=result["name"],
-                slug=result.get("slug"),
-                status=result["status"],
-                config=result["config"],
-            )
+            monitor = validator.save()
         except MonitorLimitsExceeded as e:
             return self.respond({type(e).__name__: str(e)}, status=403)
 
         ensure_cron_detector(monitor)
```

Comment:
I'm leaving this here because we won't want to run this from the detector endpoints - it's just for backwards compat on the old ones